docs/design/mono/mobile-runtimeconfig-json.md

```diff
@@ -0,0 +1,143 @@
+  </RuntimeConfigParserTask>
+</Target>
+ ```
+
+### The encoded runtimeconfig format
+
+The format is:
+1. There will be an (1- to 4-byte) ECMA-335 II.23.2 compressed unsigned integer count = N, indicating the number of key-value pairs.
+2. It will have 2xN  ECMA-335 SerString UTF8 strings (that is, each string is preceded by its length stored in the compressed unsigned integer format): Each key followed by its value.
+
+Sample input
+```
+{
+    "runtimeOptions": {
+        "configProperties": {
+            "key1": "value1",
+            "key2": "value2"
+        }
+    }
+}
+```
+
+Sample output (as hexdump -C bytes)
+```
+00000000  02 04 6b 65 79 31 06 76  61 6c 75 65 31 04 6b 65  |..key1.value1.ke|
+00000010  79 32 06 76 61 6c 75 65  32                       |y2.value2|
+00000019
+```
+
+### New embedding API entrypoint
+
+We want at least 2 ways to pass data in: either ask the runtime to open the file, or give it a pointer to the data in memory. Also, we need some way to cleanup.
+
+```
+struct MonovmRuntimeConfigArguments {
+  uint32_t kind; // 0 = Path of runtimeconfig.bin file, 1 = pointer to the blob data, >= 2 reserved
+  union {
+    struct {
+      const char *path; // null terminated absolute path
+    } name;
+    struct {
+      const char *data;
+      uint32_t data_len;
+    } data;
+  } runtimeconfig;
+};
+
+typedef void (*MonovmRuntimeConfigArgumentsCleanup)(MonovmRuntimeConfigArguments *args, void* user_data);
+
+MONO_API void
+monovm_runtimeconfig_initialize (MonovmRuntimeConfigArguments *args, MonovmRuntimeConfigArgumentsCleanup cleanup_fn, void* user_data);
+```
+
+This declaration should live in the unstable header. `monovm_runtimeconfig_initialize` should be called after `monovm_initialize` but before starting the runtime.
+
+#### Example of the usage of `monovm_runtimeconfig_initialize`
+
+```
+void
+cleanup_runtime_config (MonovmRuntimeConfigArguments *args, void *user_data)
+{
+    free (args);
+    free (user_data); // This may not be needed, depending on if there is anything needs to be freed.
+}
+
+int
+mono_droid_runtime_init (const char* executable, int managed_argc, char* managed_argv[])
+{
+  
+  ......
+
+  MonovmRuntimeConfigArguments *arg = (MonovmRuntimeConfigArguments *)malloc (sizeof (MonovmRuntimeConfigArguments));
+  arg->kind = 0;
+  arg->runtimeconfig.name.path = "path_to_generated_binary_file";
+  monovm_runtimeconfig_initialize (arg, cleanup_runtime_config, NULL);
+
+  monovm_initialize(......);
+  ......
+
+}
+```
+
+### Register and install runtime properties
+
+`monovm_runtimeconfig_initialize` will register the type `MonovmRuntimeConfigArguments` variable with the runtime. If given the path of the runtimeconfig.bin file, the runtime will read the binary data from the file, otherwise, it will parse the binary data from memory. The properties will be combined with the ones registered by `monovm_initialize` and used to initialize System.AppContext.
```

Comment:
```suggestion
`monovm_runtimeconfig_initialize` will register the type `MonovmRuntimeConfigArguments` variable with the runtime. If given the path of the runtimeconfig.bin file, the runtime will read the binary data from the file, otherwise, it will parse the binary data from memory. The properties will be combined with the ones registered by `monovm_initialize` and used to initialize System.AppContext.
```

This says `runtimeconfig.bin`, and elsewhere in the document we refer to blob. We should be consistent, and either call it "blob" or "binary" within the document based on the extension. I weakly prefer ".bin" for the extension, but it doesn't matter very much.

Reply:
yes `.bin` is the extension that I used in the test. Will update.